### DISCUSSION TITLE
Index

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#2789] [Repository Maintenance] [answered]
Regulars around here frequently want to **refer back** to old discussions. For example, when someone opens a duplicate issue, we'd like to say *which* it is a duplicate of. However, sometimes these discussions are **hard to find** via search.

Like an index in the back of a reference book, this post is intended to help **locate an issue** or comment when you **already know** what you're looking for. It _may_ also be useful as an FAQ or a list of interesting discussions to get up to date on StreetComplete development history, but that's not the primary purpose.

There are **3 _overlapping_ parts**, ordered **alphabetically**\*: **[by quest](https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403)**, by **[OSM feature](https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-1058716)** (e.g. tag), and by **[StreetComplete feature](https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646707)**.
There is also a **[Misc](https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-1116685)** section for anything else, including meta-discussion (about this index). *\*Alphabetizing is a work in progress.*

To suggest an addition or change, reply to the part you want changed. I (@smichel17) will edit it in, then your hide your comment.

**Comment by smichel17** [comment]
# By Quest…

## Building Type

### Adding the quest
- https://github.com/streetcomplete/StreetComplete/issues/25
- https://github.com/streetcomplete/StreetComplete/pull/774

### Clarifying types
- https://github.com/streetcomplete/StreetComplete/issues/1716
    - **includes _barn_**
- https://github.com/streetcomplete/StreetComplete/issues/2180
    - **includes _storage tank_**

#### Residential, especially house vs (semi)detached house vs apartment
- https://github.com/streetcomplete/StreetComplete/issues/2257
- https://github.com/streetcomplete/StreetComplete/issues/2467
- https://github.com/streetcomplete/StreetComplete/issues/2685#issuecomment-809473943

#### Abandoned/ruined buildings
- https://github.com/streetcomplete/StreetComplete/issues/2177

### Adding additional types
- https://github.com/streetcomplete/StreetComplete/issues/2265#issuecomment-727120117
- https://github.com/streetcomplete/StreetComplete/issues/3109
- https://github.com/streetcomplete/StreetComplete/issues/3078

### Buildings with multiple parts
- https://github.com/streetcomplete/StreetComplete/issues/1870

## Cycleway

#### Resurveying when previous tagging is ambiguous
- https://github.com/streetcomplete/StreetComplete/issues/2120#issuecomment-700593226

#### Remove option "No cycleway, but using the sidewalk is allowed"
- https://github.com/streetcomplete/StreetComplete/issues/2276#issuecomment-728147539

#### Making tagging "no cycleway" more obvious
- https://github.com/streetcomplete/StreetComplete/issues/2285

## Road Lanes

#### Phrasing
- https://github.com/streetcomplete/StreetComplete/issues/2284#issuecomment-729088424

#### Multi-step UX
- https://github.com/streetcomplete/StreetComplete/issues/2350

### Don't ask…
#### …for unmarked lanes:
- https://github.com/streetcomplete/StreetComplete/issues/2299

#### …in low-speed residential areas
- https://github.com/streetcomplete/StreetComplete/issues/2324

## Road Max Speed (Speed Limit)

#### Always tag `maxspeed:type` (not `source:maxspeed`) for implicit speed limits
- https://github.com/streetcomplete/StreetComplete/issues/492#issuecomment-326080750

#### Phrasing of implicit speed limits (urban/rural)
- https://github.com/streetcomplete/StreetComplete/issues/757

#### Why they are not resurveyed
- https://github.com/streetcomplete/StreetComplete/issues/1998#issuecomment-676787192

**Reply by FloEdelmann** [reply]
Add a "not listed" option: https://github.com/streetcomplete/StreetComplete/issues/3109

**Comment by smichel17** [comment]


**Reply by smichel17** [reply]
Moved to https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403

**Comment by smichel17** [comment]
Moved to https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403

**Reply by smichel17** [reply]
Moved to https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403

**Comment by smichel17** [comment]
Moved to https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403

**Reply by smichel17** [reply]
Moved to https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403

**Comment by smichel17** [comment]
# By StreetComplete Feature…

## "Other Answers" menu

### People being unaware that there are more answers in this menu
- https://github.com/streetcomplete/StreetComplete/issues/2322#issuecomment-734967835
- https://github.com/streetcomplete/StreetComplete/issues/2970#issuecomment-863314127
- https://github.com/streetcomplete/StreetComplete/issues/2973

### Too much time to reach "hide quest" option
- https://github.com/streetcomplete/StreetComplete/issues/3023

### Proposal to change menu to "can't say"
- https://github.com/streetcomplete/StreetComplete/issues/1913#issuecomment-828632563
    - It would help a little with https://github.com/streetcomplete/StreetComplete/issues/1296

## Quest Display

### Density

- https://github.com/streetcomplete/StreetComplete/issues/1874#issuecomment-645441710
- https://github.com/streetcomplete/StreetComplete/issues/2359#issuecomment-869039152
- https://github.com/streetcomplete/StreetComplete/discussions/2457#discussioncomment-267405
- https://github.com/streetcomplete/StreetComplete/issues/2944

### Order

- https://github.com/streetcomplete/StreetComplete/issues/2565#issuecomment-872584909
- https://github.com/streetcomplete/StreetComplete/commit/c1336a4ca5dbd6368d75c2c3332caeac172416ad sort quests by how easy they can be solved

### Presets (quickly toggle groups of quests)

- https://github.com/streetcomplete/StreetComplete/issues/1654
- https://github.com/streetcomplete/StreetComplete/pull/2069
- https://github.com/streetcomplete/StreetComplete/discussions/2457
- https://github.com/streetcomplete/StreetComplete/issues/2565
- :star2: https://github.com/streetcomplete/StreetComplete/issues/3034 :star2:
 
#### Won't implement since filters are better
- https://github.com/streetcomplete/StreetComplete/issues/2292

### Misc
- https://github.com/streetcomplete/StreetComplete/issues/2243
- https://github.com/streetcomplete/StreetComplete/issues/2174#issuecomment-710683825
    - Deliberately no opt-out of maintenance quests

## Quest Selection

### Answer all quests for one element

- https://github.com/streetcomplete/StreetComplete/issues/124
- https://github.com/streetcomplete/StreetComplete/issues/1969
    - Follow up: https://github.com/streetcomplete/StreetComplete/issues/2710#issuecomment-812444576

#### Multiple tags in one quest
- https://github.com/streetcomplete/StreetComplete/issues/2127

### Answer the same quest for multiple elements at once

- https://github.com/streetcomplete/StreetComplete/issues/73
- https://github.com/streetcomplete/StreetComplete/issues/2353
- https://github.com/streetcomplete/StreetComplete/issues/1303
- https://github.com/streetcomplete/StreetComplete/issues/3098#issuecomment-888255447
    - Comment has a proposed plan how this can be implemented, for interested people
- https://github.com/streetcomplete/StreetComplete/issues/2968

### Making it faster to answer specific (time-consuming) quests

- https://github.com/streetcomplete/StreetComplete/issues/1772
- https://github.com/streetcomplete/StreetComplete/issues/2542

## Resurvey

### Why some resurvey quests are not distinguishable from "normal" quests

- https://github.com/streetcomplete/StreetComplete/issues/2174#issuecomment-710683825

**Reply by smichel17** [reply]
# Quest Display

## Display density

- https://github.com/streetcomplete/StreetComplete/issues/1874#issuecomment-645441710
- https://github.com/streetcomplete/StreetComplete/discussions/2457#discussioncomment-267405
- https://github.com/streetcomplete/StreetComplete/issues/2944
- https://github.com/streetcomplete/StreetComplete/issues/2359#issuecomment-869039152

## Quest presets (quickly toggle groups of quests)

- https://github.com/streetcomplete/StreetComplete/issues/1654
    - Dupe/original: https://github.com/streetcomplete/StreetComplete/issues/1203
- https://github.com/streetcomplete/StreetComplete/discussions/2457
- https://github.com/streetcomplete/StreetComplete/pull/2069
- https://github.com/streetcomplete/StreetComplete/issues/2565
    - https://github.com/streetcomplete/StreetComplete/issues/2292
        - Won't implement since filters are better
- **Implementation:**
   https://github.com/streetcomplete/StreetComplete/issues/3034

## Misc
- https://github.com/streetcomplete/StreetComplete/issues/2243
- https://github.com/streetcomplete/StreetComplete/issues/2174#issuecomment-710683825
    - Deliberately no opt-out of maintenance quests

**Reply by smichel17** [reply]
# Quest selection

## Answer all quests for one element

- https://github.com/streetcomplete/StreetComplete/issues/124
- https://github.com/streetcomplete/StreetComplete/issues/1969
    - Follow up: https://github.com/streetcomplete/StreetComplete/issues/2710#issuecomment-812444576

## Answer multiple similar quests at once

- https://github.com/streetcomplete/StreetComplete/issues/73
- https://github.com/streetcomplete/StreetComplete/issues/2353
- https://github.com/streetcomplete/StreetComplete/issues/1303
- https://github.com/streetcomplete/StreetComplete/issues/3098#issuecomment-888255447
    - Proposed plan how this can be implemented, for interested people
- https://github.com/streetcomplete/StreetComplete/issues/2968

## Making it faster to answer specific (time-consuming) quests

- https://github.com/streetcomplete/StreetComplete/issues/2542
- https://github.com/streetcomplete/StreetComplete/issues/1772

**Reply by matkoniecz** [reply]
https://github.com/streetcomplete/StreetComplete/issues/3098#issuecomment-888255447 has proposed plan how this can be implemented for interested people

**Reply by riQQ** [reply]
Layers:
- #2461
- Discussion of the first alpha version of layers implementation #4113

**Comment by smichel17** [comment]
Merged into https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646707

**Reply by smichel17** [reply]
Moved to https://github.com/streetcomplete/StreetComplete/discussions/2789#discussioncomment-646403

**Comment by smichel17** [comment]
# By OSM Feature…
This comment is intended to be a one-stop-shop for OSM folks arriving here. It is also going to be the most work to maintain, so I'd appreciate folks suggesting links.

## Changesets

#### Are split by quest type:
- https://github.com/streetcomplete/StreetComplete/issues/21#issuecomment-292709936

#### Are not split when they contain elements which are far apart
- https://github.com/streetcomplete/StreetComplete/issues/2908#issuecomment-848330559

## Data Model

#### Quests apply only to a single element
- https://github.com/streetcomplete/StreetComplete/issues/3765#issuecomment-1040331446
- https://github.com/streetcomplete/StreetComplete/issues/3749#issuecomment-1036541910
    - https://github.com/streetcomplete/StreetComplete/discussions/2426

## Tags

#### `maxspeed:type`
- https://github.com/streetcomplete/StreetComplete/issues/492#issuecomment-326080750

#### `source:maxspeed`
- https://github.com/streetcomplete/StreetComplete/issues/492#issuecomment-326080750

**Reply by smichel17** [reply]
- Why they're split by quest type:
   https://github.com/streetcomplete/StreetComplete/issues/21#issuecomment-292709936

- Why they're not split when they're far apart
   https://github.com/streetcomplete/StreetComplete/issues/2908#issuecomment-848330559

**Comment by smichel17** [comment, marked as answer]
# Tooling / Meta

## GitHub

- https://github.com/streetcomplete/StreetComplete/discussions/3450
- https://github.com/streetcomplete/StreetComplete/discussions/3348

### Creating this post
- https://github.com/streetcomplete/StreetComplete/issues/2283.

**Reply by smichel17** [reply]
I'm thinking of restructuring this slightly. Instead of a top level post for each quest, I'd collapse it into one post where each quest is a subheading. Same with features; this would leave just 4 top-level posts (currently):

- Quests
- Features
- OSM / tags
- Tooling / meta (this one)

My reasoning is that, while the current format does lend itself to linking to, it's kind of annoying to actually scroll through this discussion and _find_ the relevant post, since there's no "alphabetical" sorting of replies.

**Reply by smichel17** [reply]
Yep, going to do this.

**Reply by smichel17** [reply]
I have been thinking about renaming this post to "GitHub Issue Index" or something like that, since it's not *just* design decisions, but any important conversation to reference. Or "Important Conversation Index"?

edit: in the end, I went with plain old "Index"
